### ISSUE TITLE
ActiveRecord's reload method doesn't always reload carrierwave attributes

### ISSUE THREAD

**Issue by x-yuri** [carrierwaveuploader/carrierwave#1997]
Steps to reproduce:

``` bash
$ rails _5.0.0_ new ra1
$ cd ra1

# Add `gem 'carrierwave', '0.11.2'` to Gemfile
$ bundle install

$ bin/rails g migration create_photoshoots url_name
$ bin/rails g migration create_photos photoshoot:references file
$ bin/rails db:migrate RAILS_ENV=test

$ bin/rails g uploader Photo
```

`app/uploaders/photo_uploader.rb`:

``` ruby
...
  def store_dir
    "uploads/#{model.photoshoot.url_name}"
  end
...
```

`app/models/photoshoot.rb`:

``` ruby
class Photoshoot < ApplicationRecord
  has_many :photos, dependent: :destroy
end
```

`app/models/photo.rb`:

``` ruby
class Photo < ApplicationRecord
  belongs_to :photoshoot
  mount_uploader :file, PhotoUploader
end
```

`test/models/photo_test.rb`:

``` ruby
require 'test_helper'

class PhotoTest < ActiveSupport::TestCase
  test "reload reloads file attribute" do
    photoshoot = Photoshoot.create url_name: 'url-name-1'
    photo = photoshoot.photos.create file: File.open('test/fixtures/files/img1.jpg')
    photoshoot.url_name = 'url-name-2'
    photoshoot.save
    photo2 = Photo.find(photo.id)
    photo2.file = File.open 'test/fixtures/files/img1.jpg'
    photo2.save
    assert_equal photo.reload.file.path, photoshoot.photos.first.file.path
  end
end
```

``` bash
$ bin/rake
...
Failure:
PhotoTest#test_the_truth [/home/yuri/_/ra1/test/models/photo_test.rb:12]:
--- expected
+++ actual
@@ -1,2 +1,2 @@
 # encoding: US-ASCII
-"/home/yuri/_/ra1/public/uploads/url-name-1/img1.jpg"
+"/home/yuri/_/ra1/public/uploads/url-name-2/img1.jpg"
```

**Comment by spodlecki** [comment]
This bug just bit me, was battling it for a while. Loading a record in console and running `asset.<version>.path` returned paths (even without file there, which is expected result), but when running tests in RSpec, `path` and `url` has a tendency to return nil if the record has not been entirely reloaded.

## Example
*uploader*
```
# ....
def processing_immediate?; end

version :something, if: :processing_immediate? do
  def full_filename(for_file)
    'something.png'
  end
end
# ....
```

*_spec.rb*
```
let(:attachment) { FactoryGirl.create(:attachment) }

# FAILS
it 'has a version' do
  expect(attachment.asset.something.path).not_to be_nil
end

# PASS
it 'has a path' do
  expect(attachment.asset.path).not_to be_nil
end
```

# FIX
*_spec.rb*
```
before(:each) do
  @id = FactoryGirl.create(:attachment).id
end
let(:attachment) { Attachment.find(id) }
```

**Comment by mshibuya** [comment]
Fixed by #1383.
